### PR TITLE
ffmpeg: add new external libs added in 7.1

### PR DESCRIPTION
**Pull request by jopejoe1**
Adds the new external libs added in FFmpeg 7.1, I have tested `lc3` and `vvenc`, ~but I could not test `lcevcdec` as I did not find a video file with that encoding or an encoder to encode into it.~ Edit also tested lcevcdec now.

lc3 and vvenc can be tested with the following:

```bash
# Testing vvc
## Encoding
ffmpeg -f lavfi -i testsrc -t 10 -c:v vvc /tmp/test.vvc
## Decoding
ffmpeg -i /tmp/test.vvc -c:v hevc /tmp/decoded-vvc.mkv

# Testing lc3
## Downloading Sample file
curl "http://fate-suite.ffmpeg.org/aac/al_sbr_cm_48_2.mp4" -o /tmp/sample.mp4
## Encoding
ffmpeg -i /tmp/sample.mp4 -ac 2 -c:a lc3 /tmp/test.lc3
## Decoding
ffmpeg -i /tmp/test.lc3 -c:a aac /tmp/decoded-lc3.aac

# Testing lcevc-dec
## Downloading Sample file
curl "https://d3mfda3gpj3dw1.cloudfront.net/vn9s0p86SVbJorX6/lcevc_vn9s0p86SVbJorX6_2.mp4" -o /tmp/lcevc-sample.mp4
## Decoding
ffmpeg -i /tmp/lcevc-sample.mp4 -filter lcevc /tmp/decoded-lcevc.mp4
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jopejoe1** [comment]
Found some samples for lcevc-dec and added information on how to test it in the description.

File: pkgs/by-name/vv/vvenc/package.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "VVENC_INSTALL_FULLFEATURE_APP" true)
+    (lib.cmakeBool "BUILD_SHARED_LIBS" (!stdenv.hostPlatform.isStatic))
+  ];
+
+  outputs = [
```
Comment (SuperSandro2000) [inline pkgs/by-name/vv/vvenc/package.nix:28]: fyi: we normally place outputs between version and src

File: pkgs/by-name/lc/lcevcdec/package.nix [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    range-v3
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeFeature "VN_SDK_FFMPEG_LIBS_PACKAGE" "")
+    (lib.cmakeBool "VN_SDK_UNIT_TESTS" false)
+    (lib.cmakeBool "VN_SDK_SAMPLE_SOURCE" false)
+  ];
+
+  outputs = [
```
Comment (SuperSandro2000) [inline pkgs/by-name/lc/lcevcdec/package.nix:85]: same here

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise lgtm

File: pkgs/by-name/vv/vvenc/package.nix (hunk 1 above)
Comment (jopejoe1) [reply pkgs/by-name/vv/vvenc/package.nix]: did not know that, thanks for the info :)

**Pull request merged**